core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -214,6 +215,9 @@ public UnicastZenPing(Settings settings, ThreadPool threadPool, TransportService
         final List<Future<TransportAddress[]>> futures =
             executorService.invokeAll(callables, resolveTimeout.nanos(), TimeUnit.NANOSECONDS);
         final List<DiscoveryNode> discoveryNodes = new ArrayList<>();
+        final Set<TransportAddress> localAddresses = new HashSet<>();
+        localAddresses.add(transportService.boundAddress().publishAddress());
```

Comment:
I think it's similar to how you had to override `TransportSerivce#boundAddress` in the tests.